### PR TITLE
HC4

### LINKED ISSUE #148: HC4
**Dataset Information:**

The HC4 collection that is accepted at ECIR 2022

<brief description>

**Links to Resources:**

https://github.com/hltcoe/HC4/tree/main/resouces/hc4

**Dataset ID(s) & supported entities:**

 - Dataset ID: hc4/{language id: zh, fa, ru}/{train, dev, test}
 - Will have {Chinese, Farsi, Russian} documents, English queries(title/description/narrative), an English report associated with each topic and qrels. 

**Checklist**

Mark each task once completed. All should be checked prior to merging a new dataset.
  
 - [x] Dataset definition (in `ir_datasets/datasets/[topid].py`)
 - [x] Tests (in `tests/integration/[topid].py`)
 - [x] Metadata generated (using `ir_datasets generate_metadata` command, should appear in `ir_datasets/etc/metadata.json`)
 - [x] Documentation (in `ir_datasets/etc/[topid].yaml`)
   - [ ] Documentation generated in https://github.com/seanmacavaney/ir-datasets.com/
 - [x] Downloadable content (in `ir_datasets/etc/downloads.json`)
   - [x] Download verification action (in `.github/workflows/verify_downloads.yml`). Only one needed per `topid`.
   - [x] ~~Any small public files from NIST (or other potentially troublesome files) mirrored in https://github.com/seanmacavaney/irds-mirror/. Mirrored status properly reflected in `downloads.json`.~~
  
**Additional comments/concerns/ideas/etc.**

The document id, qrels, and topics will be distributed through a public github repository. Users need to download the actual documents through Common Crawl. Script for downloading and validating will be provided along with the doc ids. 

The structure will be very similar to the future NeuCLIR collection. Whether these two collections will be distributed through the same repository is TBD.

### PR DESCRIPTION
**Pull request by eugene-yang**
Close #148

### CODE REVIEW

File: ir_datasets/datasets/hc4.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+class HC4Query(NamedTuple):
+    query_id: str
+    title: str
+    description: str
+    ht_title: str
+    ht_description: str
+    mt_title: str
+    mt_description: str
+    narrative: Dict[str, str]
```
Comment (seanmacavaney) [inline ir_datasets/datasets/hc4.py:83]: should be `str`?

File: ir_datasets/docs/bibliography.bib [hunk 2]
```diff
@@ -838,3 +838,15 @@ @inproceedings{Froebe2022Anchors
   title =                 {{The Power of Anchor Text in the Neural Retrieval Era}},
   year =                  2022
 }
+
+@article{Lawrie2022HC4,
+	author = {Dawn Lawrie and James Mayfield and Douglas W. Oard and Eugene Yang},
+	title = {HC4: A New Suite of Test Collections for Ad Hoc CLIR},
```
Comment (seanmacavaney) [inline ir_datasets/docs/bibliography.bib:844]: consistent tabs vs spaces?

File: ir_datasets/docs/hc4.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  Human and machine translation of the titles and descriptions in the target language (i.e., document language) are provided in the query object. 
+  (Titles and descriptions are machine-translated into all three target languages even in the laguages that they are not assessed to facillate CLIR 
+  other than English-to-X pairs, e.g., Persian-to-Chinese. Please refer to the original dataset repository for these additional resources.)
+  </li>
+  <li>Report: Each query comes with an English report that is designed to be written by professional searchers prior to the search.</li>
+  <li>Qrels: Documents are judged in three levels of relevance. Please refer to the dataset paper for the full definition of the levels. </li>
+  <li><a href="https://github.com/hltcoe/HC4">Repository</a></li>
+  <li><a href="https://arxiv.org/abs/2201.09992">Dataset Paper</a></li>
+</ul>'
+  bibtex_ids: ['Lawrie2022HC4']
```
Comment (seanmacavaney) [inline ir_datasets/docs/hc4.yaml:40]: Can you add documentation for each subset? (Doesn't need to be as detailed, but something to clarify the language used, etc.)

File: ir_datasets/etc/downloads.json [hunk 4]
```diff
@@ -3582,5 +3582,80 @@
...
+    "size_hint": 81444,
+    "expected_md5": "4c34c546c3e90de4733c3c9411ce9c6d"
+  },
+  "test/topics": {
+    "url": "https://raw.githubusercontent.com/hltcoe/HC4/main/resources/hc4/test.topics.v1-0.jsonl",
+    "size_hint": 964434,
+    "expected_md5": "a311237913a7335d45fe261e9ff7f11e"
+  },
+  "fa/docs": {
+    "instructions": "[Placeholder]",
```
Comment (seanmacavaney) [inline ir_datasets/etc/downloads.json:3604]: Fill in placeholders here with basic instructions? (These are printed to the command line when the files are not present.)

**Review by seanmacavaney — CHANGES_REQUESTED** [review]
Just a few bits and pieces

File: ir_datasets/datasets/hc4.py (hunk 1 above)
Comment (eugene-yang) [reply ir_datasets/datasets/hc4.py:83]: We actually provided one example per relevance level in the narrative field. 
I don't think there will be a lot of people using it as a query so decided to use a structure that's less common here.

Comment (seanmacavaney) [reply ir_datasets/datasets/hc4.py:83]: Ah, gotcha! That's really nice, actually. Maybe it's worth renaming then so it's clear what the keys/values mean. E.g., `narrative_by_relevane`?

**Comment by eugene-yang** [comment]
Should have fixed all the stuff listed up there. Thanks, Sean!

File: ir_datasets/datasets/hc4.py (hunk 1 above)
Comment (eugene-yang) [reply ir_datasets/datasets/hc4.py:83]: Ah that makes sense!

Comment (eugene-yang) [reply ir_datasets/datasets/hc4.py]: and done :)

File: ir_datasets/docs/hc4.yaml (hunk 3 above)
Comment (eugene-yang) [reply ir_datasets/docs/hc4.yaml:47]: Thanks, Sean!

**Pull request merged**